docs/implementation.html

```diff
@@ -22,61 +22,71 @@
     </p>
     <h3><a id="messages" href="#messages">5.2 Messages</a></h3>
     <p>
-    Messages consist of a fixed-size header, a variable length opaque key byte array and a variable length opaque value byte array. The header contains the following fields:
-    <ul>
-        <li> A CRC32 checksum to detect corruption or truncation. </li>
-        <li> A format version. </li>
-        <li> An attributes identifier </li>
-        <li> A timestamp </li>
-    </ul>
-    Leaving the key and value opaque is the right decision: there is a great deal of progress being made on serialization libraries right now, and any particular choice is unlikely to be right for all uses. Needless to say a particular application using Kafka would likely mandate a particular serialization type as part of its usage. The <code>MessageSet</code> interface is simply an iterator over messages with specialized methods for bulk reading and writing to an NIO <code>Channel</code>.
+    Messages consist of a fixed-size header, a variable length opaque key byte array and a variable length opaque value byte array. The format of the header is described in he following section.
+    Leaving the key and value opaque is the right decision: there is a great deal of progress being made on serialization libraries right now, and any particular choice is unlikely to be right for all uses. Needless to say a particular application using Kafka would likely mandate a particular serialization type as part of its usage. The <code>RecordBatch</code> interface is simply an iterator over messages with specialized methods for bulk reading and writing to an NIO <code>Channel</code>.</p>
 
     <h3><a id="messageformat" href="#messageformat">5.3 Message Format</a></h3>
+    <p>
+    Messages (aka Records) are always written in batches. The technical term for a batch of messages is a RecordBatch, and a RecordBatch contains one or more Records. In the degenerate case, we could have a RecordBatch containing a single Record.
+    RecordBatches and Records have their own headers. The format of each is described below for Kafka version 0.11.0 and later (message format version v2). <a href="https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-Messagesets">Click here</a> for details about message formats 0 and 1.</p>
+
+    <h4><a id="recordbatch" href="#recordbatch">5.3.1 Record Batch</a></h4>
+	<p> The following is the on-disk format of a RecordBatch. </p>
+	<p><pre class="brush: java;">
+		firstOffset: int64
```

Comment:
We use "baseOffset" in the code to try and avoid the suggestion that this field reflects the offset of the first record in the batch. Similarly with "baseSequence." It would be nice to have a brief section which can explain some of these details. You can borrow the documentation already written in `DefaultRecordBatch`.